### ISSUE TITLE
Crash with modified .anm file

### ISSUE THREAD

**Issue by ppobg** [thpatch/thtk#116]
I want to change all stage backgrounds to black. I extract the images with thanm from stg1bg.anm, edit them and archive them back, then create a new archive with all files with thdat. But the game crashes on entering a stage. When I unarchive the modified .anm file I get my edited images, so it's not like they aren't in there.

Crash dump:
![image](https://github.com/thpatch/thtk/assets/66318502/707ddd4b-599e-4beb-9997-92b48299579f)

**Comment by DankRank** [comment]
Did you use the [latest version](https://nightly.link/thpatch/thtk/workflows/build/master/thtk-win32-x86.zip)? It works for me. For reference, here are the commands I used.

```sh
cp th08.dat th08.dat.bak
mkdir dat
cd dat
thdat -x8 ../th08.dat
cd ..
mkdir anm
cd anm
thanm -l8 ../dat/stg1bg.anm >stg1bg.txt
thanm -x8 ../dat/stg1bg.anm
convert data/stg1bg/stg1bg.png -alpha off -fill black +opaque black tmp.png
mv tmp.png data/stg1bg/stg1bg.png
thanm -c8 ../dat/stg1bg.anm stg1bg.txt
cd ../dat
thdat -c8 ../th08.dat *
```

**Comment by ppobg** [comment]
I thought I was using the latest version from the github releases but guess not, so it was creating them with the wrong spec. Now I get this error:
```
thanm -c8 ../dat/stg1bg.anm stg1bg.txt
thanm.exe:stg1bg.txt:1,1: illegal token:  
thanm.exe:stg1bg.txt:1,1: syntax error, unexpected invalid token
```
(The token is <0xa0> according to sublime text)
[stg1bg.txt](https://github.com/thpatch/thtk/files/12642711/stg1bg.txt)

**Comment by DankRank** [comment]
Yep, the github release is pretty old.

I've tested on both Linux and Windows 10 22H2, and I couldn't reproduce the problem you have with `thanm -c8`, so idk. Your stg1bg.txt looks fine, there aren't any 0xA0 characters in it.

**Comment by ppobg** [comment]
Well replacing whitespace `[ \t]+` to a regular space in all spec files somehow fixed it and it doesn't crash anymore. The same thing happens with all my games (EoSD, IN, PCB, MoF) when creating .anm file. Kinda weird. Thanks for the help though.

**Comment by Priw8** [comment]
What shell are you using for the `-l` command? It's a known issue that certain ones (like Powershell) do funky things with encoding when redirecting text output to a file

**Comment by p-pbg** [comment]
> What shell are you using for the `-l` command? It's a known issue that certain ones (like Powershell) do funky things with encoding when redirecting text output to a file

Yes, I was using powershell for that. You're right that was the issue. Thanks
